dask/dataframe/dask_expr/diagnostics/_explain.py

```diff
@@ -82,7 +83,10 @@ def _explain_info(expr: Expr):
     }
 
 
-def _explain_details(expr: Expr):
+def _explain_details(expr: BaseExpr):
+
+    if not isinstance(expr, Expr):
+        return {}
```

Comment:
I believe explain should just work out of the box for arrays (and HLGs) but the nodes wouldn't be annotated. Still, it's something